app/v2/notifications/rest_push.py

```diff
@@ -1,59 +1,106 @@
-from flask import jsonify, request
+import json
 
-from app import authenticated_service, vetext_client, mobile_app_registry
+from celery.exceptions import CeleryError
+from jsonschema import ValidationError
+from kombu.exceptions import OperationalError
+from flask import current_app, jsonify, request
+
+from app import authenticated_service, mobile_app_registry, vetext_client
+from app.celery.provider_tasks import deliver_push
+from app.config import QueueNames
 from app.constants import PUSH_TYPE
-from app.feature_flags import FeatureFlag, is_feature_enabled
 from app.mobile_app import DEAFULT_MOBILE_APP_TYPE, MobileAppType
 from app.schema_validation import validate
 from app.utils import get_public_notify_type_text
 from app.v2.errors import BadRequestError
 from app.v2.notifications import v2_notification_blueprint
+from app.v2.dataclasses import V2PushPayload
 from app.v2.notifications.notification_schemas import push_notification_broadcast_request, push_notification_request
-from app.va.vetext import VETextBadRequestException, VETextNonRetryableException, VETextRetryableException
 
 
 @v2_notification_blueprint.route('/push', methods=['POST'])
 def send_push_notification():
-    return push_notification_helper(push_notification_request, False)
+    return push_notification_helper(push_notification_request)
 
 
 @v2_notification_blueprint.route('/push/broadcast', methods=['POST'])
 def send_push_broadcast_notification():
-    return push_notification_helper(push_notification_broadcast_request, True)
+    return push_notification_helper(push_notification_broadcast_request)
 
 
-def push_notification_helper(schema: dict, is_broadcast: bool):
+def push_notification_helper(schema: dict):
     """
     Note that this helper cannot be called other than as part of a request because it accesses
     the Flask "request" instance.
     """
-
-    if not is_feature_enabled(FeatureFlag.PUSH_NOTIFICATIONS_ENABLED):
-        raise NotImplementedError()
-
     if not authenticated_service.has_permissions(PUSH_TYPE):
         raise BadRequestError(
-            message='Service is not allowed to send {}'.format(get_public_notify_type_text(PUSH_TYPE, plural=True))
+            message='Service is not allowed to send {}'.format(
+                get_public_notify_type_text(PUSH_TYPE, plural=True),
+            ),
+            status_code=403,
```

Comment:
I don't understand why this isn't 400.  If it should be 403, it seems like we should be raising a different exception.

Reply:
It is currently a 400. A 403 status code means access to a requested resource is forbidden. It occurs when a server understands a request but refuses to authorize it. In this case, that seems more appropriate (the service is not authorized to send push).

Reply:
@cris-oddball Thank you for responding to Dave

Reply:
I concur that 403 makes more sense.  I'm suggesting that we should create a new exception, perhaps "PermissionDeniedError," rather than hack the BadRequestError.  A bad request is 400.

Reply:
ooh I gotcha. Yea, the BadRequestError threw me too. PermissionDeniedError seems better.

Reply:
We do not use `PermissionDeniedError` anywhere in this code base - For 403 errors we generally use `InvalidRequest`, `AuthError`, or ` BadRequestError`. I'm hesitant to add error type to this code base

Reply:
I have no issues with the `BadRequestError`. The class allows for any status code. The name of it could have been more narrow or `InvalidRequest` could have more derived classes but I don't think any of that is necessary for this ticket.